controllers/challenge_controller_test.go

```diff
@@ -0,0 +1,223 @@
+package controllers_test
+
+import (
+	"github.com/kamva/mgm/v3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"net/http"
+
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/drift-org/backend/controllers"
+	"github.com/drift-org/backend/helpers"
+	"github.com/drift-org/backend/models"
+)
+
+var _ = Describe("ChallengeController", func() {
+
+	var (
+		challengeController controllers.ChallengeController = controllers.NewChallengeController()
+	)
+
+	Describe("Register", func() {
+		It("Basic & Unique Task Name/Location", helpers.TestWithMongo("ChallengeController-Create-Unique", func() {
+
+			// Create a sample request for a challenge to be created.
+			context := helpers.CreateTestContext("", `{
+				"latitude": 34.0701449,
+				"longitude": -118.4422936,
+				"address": "1234 Main Street",
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 10
+			}`)
+
+			// Test the first challenge, and that the response is OK.
+			challengeController.Create(context)
+			Expect(context.Writer.Status()).To(Equal(http.StatusOK))
+
+			// Test that the challenge was written to mongo.
+			challenge := &models.Challenge{}
+			_ = mgm.Coll(challenge).First(bson.M{}, challenge)
+			Expect(challenge).NotTo(BeNil())
+
+			// Attempt creating a user with just the same location.
+			context2 := helpers.CreateTestContext("", `{
+				"latitude": 34.0701449,
+				"longitude": -118.4422936,
+				"address": "1234 Main Street",
+				"taskName": "Task 2",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context2)
+			Expect(context2.Writer.Status()).To(Equal(http.StatusOK))
+
+			// Test that the challenge was written to mongo.
+			_ = mgm.Coll(challenge).First(bson.M{}, challenge)
```

Comment:
I don't think this is doing what you intended? This is testing if there are any challenges in the collection, but you already had one from the challenge before. You should add to the query to make sure that you are testing that the correct challenge (the one with name "Task 2") was added.

Reply:
fixed